content/blog/2016/2016-08-29-jenkins-world-speaker-blog-goodgame.adoc

```diff
@@ -0,0 +1,42 @@
+---
+layout: post
+title: "Enforcing Jenkins Best Practices"
+tags:
+- event
+- JenkinsWorld
+author: hinman55
+---
+
+NOTE: This is a guest post by Jenkins World speaker David Hinske, Release Engineer at link:https://www.goodgamestudios.com/[Goodgame Studios].
+
+image:/images/post-images/Jenkins-World-logo.png [role=right]
+
+Hey there, my name is David Hinske and I work at Goodgame Studios (GGS), a game development company in Hamburg, Germany. As Release Engineer in a company with several development teams, it comes in handy using several Jenkins-instances. While this approach works fine in our company and gives the developers a lot of freedom, we came across some long-term problems concerning maintenance and standards. Problems, which where mostly caused by misconfiguration/non-usage of plugins. With “configuration as code” in mind, I took the approach to apply static code-analysis with the help of SonarQube, a platform to manage code quality, for all of our Jenkins-Job-configuration.
+
+As a small centralized team, we were looking for an easy way to control the health of our growing Jenkins-infrastructure. With considering “configuration as code“, I developed a simple extension of SonarQube, to manage the quality and usage of all spawned Jenkins-instances. The given SonarQube-features (like customized rules/metrics, quality profiles and dashboards) allows us and the development-teams to analyze and measure the quality of all created jobs in our company. Even though Jenkins configuration analysis can not cover all SonarQubes axes of code quality, I think there is still potential for conventions/standards, duplications, complexity, potential bugs (misconfiguration) and design and architecture.
```

Comment:
"Jenkins infrastructure"
"Jenkins instances"
"SonarQube features"
"development teams"

s/allows us/allow us/ (because 'features' is plural)
s/can not/cannot/
s/SonarQubes axes/SonarQube's axes/ (I'm guessing this should be possessive)